qiskit_ibm_runtime/estimator.py

```diff
@@ -0,0 +1,111 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Estimator primitive."""
+
+from typing import Iterable, Optional, Dict, Sequence, Any
+
+from qiskit.circuit import QuantumCircuit, Parameter
+from qiskit.quantum_info import SparsePauliOp
+
+# TODO import BaseEstimator and EstimatorResult from terra once released
+from .qiskit.primitives import BaseEstimator, EstimatorResult
+from .ibm_runtime_service import IBMRuntimeService
+from .runtime_session import RuntimeSession
+
+
+class Estimator(BaseEstimator):
+    """Estimator primitive."""
+
+    def __init__(
+        self,
+        circuits: Iterable[QuantumCircuit],
+        observables: Iterable[SparsePauliOp],
+        parameters: Optional[Iterable[Iterable[Parameter]]] = None,
+        transpile_options: Optional[Dict] = None,
+        service: Optional[IBMRuntimeService] = None,
+        backend_name: Optional[str] = None,
```

Comment:
can we make this `backend: Optional[str | Backend]`?

Reply:
If we make this class directly instantiable as Ali suggests, then we can add Backend here, otherwise just a string would do. We need to resolve that discussion first.

Reply:
As discussed on call with @ajavadia and @levbishop, it was decided that we don't rename classes as Ali suggested and that this class won't be directly instantiated by user, so backend could be just a string here. IBMEstiimator class takes `string` or `IBMBackend`.

Reply:
Ok, I am good with that.